source/server/watchdog_impl.h

```diff
@@ -19,26 +19,23 @@ class WatchDogImpl : public WatchDog {
   /**
    * @param interval WatchDog timer interval (used after startWatchdog())
    */
-  WatchDogImpl(Thread::ThreadId thread_id, TimeSource& tsource, std::chrono::milliseconds interval)
-      : thread_id_(thread_id), time_source_(tsource),
-        latest_touch_time_since_epoch_(tsource.monotonicTime().time_since_epoch()),
-        timer_interval_(interval) {}
+  WatchDogImpl(Thread::ThreadId thread_id, std::chrono::milliseconds interval)
+      : thread_id_(thread_id), timer_interval_(interval) {}
 
   Thread::ThreadId threadId() const override { return thread_id_; }
-  MonotonicTime lastTouchTime() const override {
-    return MonotonicTime(latest_touch_time_since_epoch_.load());
+  uint64_t touchCount() const {
+    return touch_count_.load();
   }
 
   // Server::WatchDog
   void startWatchdog(Event::Dispatcher& dispatcher) override;
   void touch() override {
-    latest_touch_time_since_epoch_.store(time_source_.monotonicTime().time_since_epoch());
+    ++touch_count_;
   }
 
 private:
   const Thread::ThreadId thread_id_;
-  TimeSource& time_source_;
-  std::atomic<std::chrono::steady_clock::duration> latest_touch_time_since_epoch_;
+  std::atomic<uint64_t> touch_count_ = 0;
```

Comment:
Don't we support 32 bit ARM? Will this work there?
I sort of expect the std::atomic vs absl::Mutex discussion here.

Reply:
Can you provide more context?  I assume ARM supports atomic ops.  Are they particularly expensive in ARM?

Reply:
It would be acceptable to make this a 32 bit type, wrap around behavior would be acceptable since we're just checking that the numbers are different.  Another implementation option would be to have the guarddog reset the counter to 0 when stepping through which should prevent overflow.